### PR TITLE
skip email validation on empty string

### LINKED ISSUE #13587: 500 on org private to public, public to private and public to limited
- Gitea version (or commit ref): `Gitea version 1.14.0+dev-224-g586bfb9f3 built with GNU Make 4.3, go1.15.5 : bindata, sqlite, sqlite_unlock_notify`
- Git version: `git version 2.29.2.454.gaff20da3a2`
- Operating system: Arch Linux
- Built from
  - [x] source
- Run with
  - [x] systemd
- Database:
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes

## Description
After setting a private org to public I am met with a 500 (verified both locally and on try.gitea.io).
The same is true for public org set to private or public org set to limited.

https://try.gitea.io/nuoneplstest500
https://try.gitea.io/nuoneplspublic500

links to short videos of the issue
http://ix.io/2Eoa

### LINKED ISSUE #13653: 500 with "UpdateUser: e-mail invalid [email: ]" when updating org
- Gitea version (or commit ref): latest master

## Description

When updating a org like https://try.gitea.io/org/silverwind-org/settings and trying to set description or full name, a 500 is seen on save and the org settings are not saved. In the log I see 

````
...uters/org/setting.go:94:SettingsPost() [E] UpdateUser: e-mail invalid [email: ]
````

The form does not contain a email field. Probably affects 1.13 and 1.14.

### PR DESCRIPTION
**Pull request by 6543**
- move validation into its own function
- use a session for UpdateUserSetting

close #13587
close #13653

### CODE REVIEW

**Review by lunny — APPROVED** [review]

**Comment by 6543** [comment]
regression of #13475 - so I'll add skip-changelog label

**Review by lafriks — APPROVED** [review]

File: models/user_mail.go [hunk 1]
```diff
@@ -33,6 +33,17 @@ type EmailAddress struct {
 	IsPrimary   bool `xorm:"-"`
 }
 
+// ValidateEmail check if email is a allowed address
+func ValidateEmail(email string) error {
+	if len(email) != 0 {
```
Comment (lafriks) [inline models/user_mail.go]: It would be better to inverse this if and return nil if len == 0

**Review by lafriks — CHANGES_REQUESTED** [review]
As per my comment

**Comment by 6543** [comment]
@lafriks done

**Review by lafriks — APPROVED** [review]

**Pull request merged**